src/libraries/System.Private.CoreLib/src/System/Buffers/Binary/ReaderBigEndian.cs

```diff
@@ -48,12 +48,9 @@ public static Half ReadHalfBigEndian(ReadOnlySpan<byte> source)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static short ReadInt16BigEndian(ReadOnlySpan<byte> source)
         {
-            short result = MemoryMarshal.Read<short>(source);
-            if (BitConverter.IsLittleEndian)
-            {
-                result = ReverseEndianness(result);
-            }
-            return result;
+            return BitConverter.IsLittleEndian ?
+                ReverseEndianness(MemoryMarshal.Read<short>(source)) :
```

Comment:
The JIT is reordering the IR so that the `ReverseEndianness` occurs directly over the `Unsafe.ReadUnaligned` and emitting just `movbe reg, [addr]`? The throw helpers/checks and inlining boundaries aren't getting in the way here?

Reply:
Yes, thats the case. We end up generating:

```asm
mov      rax, bword ptr [rcx]
mov      ecx, dword ptr [rcx+8]
cmp      ecx, 4
jl       SHORT G_M53413_IG04
movbe    eax, dword ptr [rax]
```